### DISCUSSION TITLE
Limiting bot access

### DISCUSSION THREAD

**Discussion by DannyBen** [shardlab/discordrb#116] [Q&A] [answered]
Hi,

I am trying to figure out how do I limit access to the bot commands only to Admins.

I am using the `CommandBot` class, and the only thing I found is [this Wiki example](https://github.com/shardlab/discordrb/wiki/Commands#command-attributes) which describes `permission_level` on a per-command bases.

Is there a global setting?

On a side note - I was wondering why the slash commands example is so much more complex than the non-slash commands example, ideally I am looking to do slash commands - as it seems that most bots move to it, and we get the nice autocomplete.

**Comment by Droid00000** [comment, marked as answer]
I'm really late to the party, but I'll try and give an answer:

1. Tentatively, I don't think there's a global permission setting that applies to all commands as a default for CommandBot, but I may be wrong.

2. Slash commands are more complex due to how Discord designed them. We've done whatever we can in order to make the experience as easy as possible, but a certain degree of complexity is always going to be present.
